leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/LeshanClientBuilder.java

```diff
@@ -123,6 +127,16 @@ public LeshanClientBuilder setDtlsConfig(DtlsConnectorConfig.Builder config) {
         return this;
     }
 
+    public LeshanClientBuilder noUnsecureEndpoint() {
```

Comment:
disableSecureEndpoint() and disableUnsecureEndpoint()?
I think some Javadoc could be helpful here, to explain that localAddress/dtlsConfig may be ignored if the endpoints are disabled.